File: front_end/types/recipe.ts
export interface User {
  username: string;
  profile_image: string;
}

export interface Recipe {
  id: number;
  title: string;
  description: string;
  thumbnail: string;
  category: Category;
  prep_time: string;
  user: User;
  isPaid: boolean;
}

export interface RecipeDetail {
  user:               User;
  category:           Category;
  category_id:        number;
  description:        string;
  id:                 number;
  created_at:         Date;
  is_paid:            boolean;
  prep_time:          number;
  price:              number;
  recipe_images:      RecipeImage[];
  steps:              Step[];
  recipe_ingredients: Ingredient[];
}

export interface RecipeImage {
  image_url: string;
}

export interface Filters {
  categories: Category[];
  prepTime: string;
  creator: string;
}

export interface Category {
  id: number;
  name: string;
  description: string;
}

export interface Creator {
  id: number;
  name: string;
  avatar: string;
  recipeCount: number;
}

export interface Ingredient {
  id: number;
  name: string;
  amount: string;
  unit: string;
}

export interface Step {
  id: number;
  step_number: number;
  instruction: string;
  image?: string;
}

export interface Comment {
  id: number;
  user: User;
  content: string;
  createdAt: string;
}

// export interface Recipe {
//   id: number;
//   title: string;
//   description: string;
//   image: string;
//   category: string;
//   prepTime: string;
//   cookTime: string;
//   servings: number;
//   difficulty: "Easy" | "Medium" | "Hard";
//   author: Author;
//   isPaid: boolean;
//   ingredients: Ingredient[];
//   steps: Step[];
//   comments: Comment[];
//   likes: number;
//   isLiked?: boolean;
//   isBookmarked?: boolean;
//   rating: number;
//   userRating?: number;
// }

File: services/models/recipe.go
package models

// Ingredient represents the input for an ingredient in the recipe.

type Recipe_ingredients struct {
	Ingredient Ingredient `form:"ingredient"`
	Amount     string     `form:"amount"`
	Unit       string     `form:unit`
}

type Ingredient struct {
	Name string `form:"name"`
}

// RecipeImage represents the input for an image associated with a recipe step or the recipe itself.
type RecipeImage struct {
	ImageURL string `form:"image_url"`
}

// Step represents the steps in the recipe, including instructions and associated images.
type Step struct {
	StepNumber   int         `form:"step_number"`
	Instruction  string      `form:"instruction"`
	RecipeImages RecipeImage `form:"recipe_images"`
}

// FileUploadInput represents the input structure for the fileUpload mutation.
type Recipe struct {
	Title       string               `form:"title"`
	Description string               `form:"description"`
	PrepTime    int                  `form:"prepTime"`
	Thumbnail   string               `form:"thumbnail"`
	IsPaid      bool                 `form:"is_paid"`
	Price       float64              `form:"price"`
	UserID      int                  `form:"user_id"`
	CategoryID  int                  `form:"category_id"`
	Ingredients []Recipe_ingredients `form:"ingredients[]"`
	Steps       []Step               `form:"steps[]"`
	Images      []RecipeImage        `form:"images[]"`
}

type InsertRecipesOne struct {
	ID        int    `json:"id"`
	Title     string `json:"title"`
	CreatedAt string `json:"created_at"`
}
